File: kotlin/src/main/kotlin/AlternatingGroups2_3208.kt

fun numberOfAlternatingGroups(colors: IntArray, k: Int): Int {
	var result = 0
	var counter = k - 1
	var previousColor = colors[0]
	var currentColor: Int
	for (i in 1 until colors.size + k - 1) {
		currentColor = colors[i % colors.size]
		if (currentColor != previousColor) {
			counter--
		} else {
			counter = k - 1
		}

		previousColor = currentColor

		if (counter <= 0) {
			result++
		}
	}

	return result
}

File: python/number_of_alternating_groups_3208/alternating_groups_2_3208.py
from typing import List


class Solution:
    def numberOfAlternatingGroups(self, colors: List[int], k: int) -> int:
        result = 0
        counter = k -1
        previous_color: int = colors[0]
        current_color: int
        for i in range(1, len(colors) + k - 1):
            current_color = colors[i % len(colors)]
            if current_color != previous_color:
                counter -= 1
            else:
                counter = k - 1

            previous_color = current_color
            if counter <= 0:
                result += 1

        return result

